### PR TITLE
DCOS-48552: add the ability to disable CheckboxTable rows

### PR DESCRIPTION
**Pull request by mperrotti**
Added the ability to set CheckboxTable rows to be non-selectable and/or visually muted

screen shot 2019-03-04 at 3 57 55 pm

### CODE REVIEW

File: packages/table/components/CheckboxTable.tsx [hunk 1]
```diff
@@ -109,20 +139,26 @@ class CheckboxTable extends React.PureComponent<
       this.setState({ headerChecked: e.target.checked });
     };
 
+    console.log(`Boolean(selectedLength): ${Boolean(selectedLength)}`);
+    console.log(
+      `selectedLength === maxSelectedLength: ${selectedLength ===
+        maxSelectedLength}`
+    );
+    console.log(`this.state.headerChecked: ${this.state.headerChecked}`);
```
Comment (natmegs) [inline packages/table/components/CheckboxTable.tsx]: 🎨

File: packages/table/stories/CheckboxTable.stories.tsx [hunk 2]
```diff
@@ -151,4 +151,43 @@ storiesOf("Table/CheckboxTable", module)
       />
     </CheckboxTableHelper>
   ))
+  .addWithInfo("w/ disabled row", () => (
+    <CheckboxTableHelper
+      data={items}
+      disabledRows={{ [items[0].name]: true }}
```
Comment (GeorgiSTodorov) [inline packages/table/stories/CheckboxTable.stories.tsx]: This seems like an unnecessarily complex data structure to me. Can we do something like
`disabledRows = [items[0].name, items[1].name]`?

Comment (mperrotti) [reply packages/table/stories/CheckboxTable.stories.tsx]: I wanted to be consistent with the data structure for `selectedRows`. Or do you think it's so complex that it's not worth being consistent with `selectedRows`?

Comment (GeorgiSTodorov) [reply packages/table/stories/CheckboxTable.stories.tsx]: In that case I prefer consistency then.

**Pull request merged**